addons/stock/models/stock_quant.py

```diff
@@ -130,15 +130,7 @@ def _set_inventory_quantity(self):
     def _search_on_hand(self, operator, value):
         if operator not in ['=', '!='] or not isinstance(value, bool):
             raise UserError(_('Operation not supported'))
-        domain_loc = []
-        if self.env.context.get('active_model') == 'product.template':
-            product_tmpl = self.env['product.template'].browse(self.env.context.get('active_id'))
-            domain_loc = product_tmpl.product_variant_ids._get_domain_locations()[0]
-        elif self.env.context.get('active_model') == 'product.product':
-            product = self.env['product.product'].browse(self.env.context.get('active_id'))
-            domain_loc = product._get_domain_locations()[0]
-        else:
-            return domain_loc
+        domain_loc = self.env['product.product']._get_domain_locations()[0]
```

Comment:
don't you, at least, filter on internal location ?

Reply:
that's the job of _get_domain_locations